protocols.py

```diff
@@ -0,0 +1,19 @@
+from shared_types import MethodList
+
+def generate_protocol_name(class_name: str, method_name: str) -> str:  
+  return f'{class_name}_{"".join([name_component[0].upper() + name_component[1:] for name_component in method_name.split("_")])}'
+
+def write_protocols(method_list: MethodList) -> None:
+  with open('main.py', 'a+') as file: # TODO: Write Protocols in the proper file
+    for method_name, method in method_list:
+      class_name = method.__self__.__class__.__name__
+      protocol_name = generate_protocol_name(class_name, method_name)
+      
+      file.write(f'''
```

Comment:
Great use of an f-string here. I recommend using using a `\` character after the initial opening triple-quotes to avoid having the extra newline show up.

```suggestion
      file.write(f'''\
```